### ISSUE TITLE
Windows support?

### ISSUE THREAD

**Issue by lucasjinreal** [qiuyu96/CoDeF#17]
Hello, tinycudann are really hard to install on windows, at least I failed on cuda11.8 + windows.
Would consider add vanilla pytorch in modeling so that users can use without tinycudann?

**Comment by ken-ouyang** [comment]
Hi. You may disable the option deform_hash and vid_hash for training if you do not have tiny-cuda. We implement a version with vanilla pytorch using positional encoding. But the training time will greatly increase and thus it is not recommended.

**Comment by lucasjinreal** [comment]
inference also not support

**Comment by lucasjinreal** [comment]
disable deform_hash and vid_hash will effect result performance?

**Comment by ken-ouyang** [comment]
Certainly, the type of motion present within a video sequence can impact the performance of the model to varying degrees. Specifically, if the video primarily contains rigid motion, you can expect the performance to remain relatively consistent.

**Comment by justinjohn0306** [comment]
How do I switch to gloo backend?
![Screenshot 2023-08-27 222904](https://github.com/qiuyu96/CoDeF/assets/34035011/81ed413e-7aa4-4edf-a68b-3d0fe3c47776)

**Comment by justinjohn0306** [comment]
Done.

![Screenshot 2023-08-27 233306](https://github.com/qiuyu96/CoDeF/assets/34035011/2ae8e972-2ced-41eb-b2b3-a2eb968e360b)

**Comment by ataraxyz** [comment]
How did you switch? Can you post a how to?

**Comment by yuanjiacheng** [comment]
> How did you switch? Can you post a how to?

replace strategy="ddp_find_unused_parameters_true" in train.py by
from pytorch_lightning.strategies import DDPStrategy
ddp_gloo = DDPStrategy(process_group_backend="gloo",find_unused_parameters=True)
trainer = Trainer(max_steps=hparams.num_steps,
                  precision=16 if hparams.vid_hash == True else 32,
                  callbacks=[checkpoint_callback],
                  logger=logger,
                  accelerator='gpu',
                  devices=hparams.gpus,
                  num_sanity_val_steps=1,
                  benchmark=True,
                  profiler="simple" if len(hparams.gpus) == 1 else None,
                  val_check_interval=hparams.valid_iters,
                  limit_val_batches=hparams.valid_batches,
                  strategy=ddp_gloo)
